src/btree/bt_page.c

```diff
@@ -346,7 +369,14 @@ __inmem_col_var(
 	 */
 	indx = 0;
 	cip = page->pg_var;
-	WT_CELL_FOREACH_BEGIN(session, btree, page->dsk, unpack, true) {
+	WT_CELL_FOREACH_BEGIN(session, btree, page->dsk, unpack) {
+		/* Optionally skip unstable values */
+		if (check_unstable &&
```

Comment:
> Could you push the `check_unstable` parameter inside `__unstable_skip` and also pass page directly and do the decrement of `entries` in the function? That would reduce the code duplication amongst callers of `__unstable_skip`

I left the code this way for two reasons. First, I think `__unstable_skip()` may become more complicated in the future (for example, if we were to start skipping address blocks in internal pages for some reason), and I'd prefer to keep its functionality strictly limited for now. Second, I'm unhappy with the decrement of `WT_PAGE.entries`, I'm concerned the inaccuracy this implies in page counts isn't OK and it will need adjustment, in which case those fixes would have to be in the code above this function, not `__unstable_skip()`.

In summary, the structure was deliberate, but I don't have particularly good reasons for the decision  -- if you feel strongly, happy to make the change.

Reply:
I don't feel strongly - since you had a reason I'll resolve.